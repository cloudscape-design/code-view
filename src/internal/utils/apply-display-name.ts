export function applyDisplayName<T>(component: T, displayName: string): void {
  (component as any).displayName = displayName;
}
