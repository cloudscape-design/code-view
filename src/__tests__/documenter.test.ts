// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import { expect, test } from "vitest";
import { requireComponentDefinition } from "./utils";

test(`definition for code-view matches the snapshot`, () => {
  const definition = requireComponentDefinition("code-view");
  expect(definition).toMatchSnapshot("code-view");
});
