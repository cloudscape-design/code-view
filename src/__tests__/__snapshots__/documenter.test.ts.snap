// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`definition for code-view matches the snapshot 1`] = `
{
  "events": [],
  "functions": [],
  "name": "CodeView",
  "properties": [
    {
      "description": "Adds an \`aria-label\` to the component. Use this label when there is not enough context around the code snippet to describe its purpose or content.",
      "name": "ariaLabel",
      "optional": true,
      "type": "string",
    },
    {
      "description": "Adds \`aria-labelledby\` to the component. Use this property to reference the ID of an existing element that provides a descriptive label for the code snippet.",
      "name": "ariaLabelledby",
      "optional": true,
      "type": "string",
    },
    {
      "description": "The code content to be displayed.",
      "name": "content",
      "optional": false,
      "type": "string",
    },
    {
      "description": "A function to perform custom syntax highlighting.",
      "name": "highlight",
      "optional": true,
      "type": "(code: string) => React.ReactNode",
    },
    {
      "description": "Controls the display of line numbers.
Defaults to \`false\`.
",
      "name": "lineNumbers",
      "optional": true,
      "type": "boolean",
    },
    {
      "description": "Controls the number the line numbers start counting from.
Defaults to \`1\`.
",
      "name": "lineNumbersStart",
      "optional": true,
      "type": "number",
    },
  ],
  "regions": [
    {
      "description": "An optional slot to display a button to enable users to perform actions, such as copy or download the code snippet.",
      "isDefault": false,
      "name": "actions",
    },
  ],
  "releaseStatus": "stable",
}
`;
